Exercise 2: Batch and Scheduled Jobs

```diff
@@ -0,0 +1,71 @@
+OpportunitySummaryBatch
+
+public class OpportunitySummaryBatch implements Database.Batchable<sObject> {
+    public Database.QueryLocator start(Database.BatchableContext BC) {
+        Date startDate = Date.today().toStartOfMonth();
+        Date endDate = startDate.addMonths(1).toStartOfMonth();
+        return Database.getQueryLocator([SELECT Id, Amount, StageName, AccountId FROM Opportunity WHERE CreatedDate >= :startDate AND CreatedDate < :endDate]);
+    }
+    
+ public void execute(Database.BatchableContext BC, List<Opportunity> scope) {
```

Comment:
too much logic in a batchable class. please move the main logic to a separated apex class handler